### ISSUE TITLE
Add feature to insert lines/nodes in rebase editor

### ISSUE THREAD

**Issue by dlech** [eamodio/vscode-gitlens#1268]
There are currently some rebasing features that cannot be used in the current rebase editor.

```
# Commands:
# p, pick <commit> = use commit
# r, reword <commit> = use commit, but edit the commit message
# e, edit <commit> = use commit, but stop for amending
# s, squash <commit> = use commit, but meld into previous commit
# f, fixup <commit> = like "squash", but discard this commit's log message
# x, exec <command> = run command (the rest of the line) using shell
# b, break = stop here (continue rebase later with 'git rebase --continue')
# d, drop <commit> = remove commit
# l, label <label> = label current HEAD with a name
# t, reset <label> = reset HEAD to a label
# m, merge [-C <commit> | -c <commit>] <label> [# <oneline>]
# .       create a merge commit using the original merge commit's
# .       message (or the oneline, if no original merge commit was
# .       specified). Use -c <commit> to reword the commit message.
```

The current GitLens editor allows pick, reword, edit, squash, fixup and drop. And it only allows these on existing commits in the rebase.

One thing I occasionally do is essentially rebase and cherry-pick at the same time. To do this, I just do a rebase and then add an extra `pick` line to cherry-pick a commit from a different branch.

In order to do this with the GitLens editor, it would need to be modified to be able to add/insert nodes/lines with an arbitrary commit hash.

Another feature I use somewhat often is the `x`/`exec` command. This could be implemented in the GitLens editor by adding a new node/line type that takes arbitrary text instead of the commit hash as the argument. The same feature described above for addding/inserting lines could be used to add lines of this type.

In other words, given the following input:

```
pick aaaaaa message
pick bbbbbb message
pick cccccc message
```

I would like GitLens to be able to change it to:

```
pick aaaaaa message
exec sed 's/first/second/' -i some-file
exec git commit -a --amend --no-edit
pick bbbbbb message
pick dddddd message
pick cccccc message
```

**Comment by dlech** [comment]
Also fixing #1190 would be nice in this regard since it probably makes more sense to most people that the `exec` commands would be run in top to bottom order rather than bottom to top.

**Comment by Swastik2000** [comment]
Can I take this issue?
